### ISSUE TITLE
Colours are not calculated from the 0 when using .update()

### ISSUE THREAD

**Issue by Waghabond** [kurkle/chartjs-plugin-autocolors#20]
There is an issue where if .update() is called on the chart to update datasets, the colours are not recalculated from offset=0. instead the colours begin to look more an more similar the more time .update() is called in the chart without a page reload. The prefered behaviour in this case should probably be to reset offset when .update() is called.

**Comment by kurkle** [comment]
Thanks for the report @Waghabond.

What version are you using? Assuming 0.0.7, does reverting to 0.0.6 fix the issue?

**Comment by Waghabond** [comment]
Hey thanks for the quick reply,

Yes i was indeed using 0.0.7, I've attempted downgrading to 0.0.6 but the issue continues to persist. 
Some further details about code:

The chart is initialised like so:
```js
this.chart = new Chart(this.canvasTarget, {
    data: {
        datasets: await this.getData(),
    },
    options: {
        responsive: true,
        scales: {
            x: {
                type: "timeseries",
                time: {
                    unit: "day",
                },
            },
        },
        plugins: {
            // @ts-expect-error
            autocolors: { mode: "dataset" },
        },
    },
})
```
and then later updates are performed like so:
```js
const data = await this.getData()
this.chart.data.datasets = data
this.chart.update()
```
A workaround I've found is to simply destroy and recreate the chart in order to perform the "update", however this causes the chart to visually disappear for a split second and reappear which is not ideal and the dom shifts around quickly and look a bit strange for that split second (not a huge problem of course and the work around is okay for now).

So for some further infor about my environment: 
Laravel 9.35.1: Backend.
Chart.js 3.9.1.
chartjs-plugin-autocolors 0.0.6/0.0.7.
@hotwired/stimulus 3.1.0: Using this as a way to simplify JS event handling.
Google Chrome 106.0.5249.121.

Let me know if you require any further info!
Cheers =)

BTW, while I have you here i was also wondering if there are any plans to add typescript stubs to the package? It would be an amazing quality of life improvement especially when considering that chart.js itself has very comprehensive type stubs. I know it's a lot of work so its no big deal if not.

**Comment by kurkle** [comment]
So, seems you are using a framework of some kind. Vue perhaps?

Typescript typings should not be a big effort to add. I'll fork that to another issue.

**Comment by Waghabond** [comment]
No, its just html, css, and js. 
Stimulus is sometimes called a framework, but its mostly just a way of writing es6 javascript classes and easily setting up methods on the classes as event handlers. 
Eg. the above code snippets are from the initialize() and update() methods of a class that I have created. The `this.canvasTarget` is the DOM canvas element that the chart is created on.

**Comment by kurkle** [comment]
Could you try creating a simple reproduction in codepen or similar tool?

**Comment by Waghabond** [comment]
sorry for going silent for a bit, i couldnt recreate the issue on code pen, i might try create a git repo so i can add these packages with yarn instead of CDNs. I'll keep you posted!

**Comment by MorningK** [comment]
I have met this issue too. I am using chartjs and [react-chartjs-2](https://react-chartjs-2.js.org/components/chart), i try to set redraw prop to true, and the problem seems to be resolved. but setting redraw to true seems takes a lot of resources.

**Comment by kurkle** [comment]
This can be mitigated by doing `myChart._autocolor = undefined` before `myChart.update()`

I think this could be made official by requiring a `chart.reset()` instead. Or just an option to reset the plugin on every chart.update().

**Comment by Waghabond** [comment]
Personally I think the plugin should just reset itself on a chart.update every time. it doesn't make sense for the colours to become more similar every time the chart updates

**Comment by Waghabond** [comment]
when you recolour every thing from the beginning, wouldnt the end result be the same regardless of if data was added or completely replaced?

**Comment by kurkle** [comment]
Not for datasets that already have colors
https://github.com/kurkle/chartjs-plugin-autocolors/blob/24580e734fd4c22f183e0c4517d58f97c20c8584/src/index.js#L60-L63

This was changed in #16 

I guess in #16 the dataset objects were persisted across updates, thus preserving the already set colors.
A simple fix would be to skip a color for those datasets that already have a color (and it matches the current color).

Anyway, seems like its time to add tests.

**Comment by kurkle** [comment]
fixed in 6a34c0b496637213915c4e783e6c55eced388f75
included in v0.1.0 (released in couple of minutes)
